src/Pages/HomePage/HomePage.js

```diff
@@ -1,12 +1,30 @@
-import React from 'react';
+import axios from 'axios';
+import React, {useEffect, useState} from 'react';
+import { Link } from 'react-router-dom';
 import CategoryCard from '../../Components/CategoryCard/CategoryCard';
+import Loader from '../../Components/Loader/Loader'
 import './HomePage.css'
 
 function HomePage() {
 
-  const categories = [{id: 1, imgurl: '/Images/video.jpg', CategoryName: 'Editing'}, 
-  {id: 2, imgurl: '/Images/video.jpg', CategoryName: 'Camera'}, {id: 3, imgurl: '/Images/video.jpg', CategoryName: 'Shooting'},
-  {id: 4, imgurl: '/Images/video.jpg', CategoryName: 'Videography'}]
+  const [categories, setcategories] = useState([])
+  const [isloading, setisloading] = useState(false)
+
+  const getCategories= async () => {
+    try{
+      let categoryresult = await axios.get('/api/categories')
+      setcategories(categoryresult.data.categories)
+      setisloading(false)
+    }catch(err){
+      console.log(err)
+      setisloading(false)
+    }
```

Comment:
instead of doing "setisloading(false)" twice in try and catch you can add this inside finally block.